sign_up/lib/homepage.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:flutter/material.dart';
+import 'constants.dart';
+
+class HomePage extends StatefulWidget {
+  const HomePage({super.key});
+
+  @override
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  bool isObscure = true;
+  final emailController = TextEditingController();
+  final passwordController = TextEditingController();
+  String? errorTextEmail;
+  String? errorTextPassword;
+  final nodePassword = FocusNode();
+  String textPassword = 'Must have at least 8 characters';
+
+  String? errorTextEmailOn() {
+    if (emailController.value.text.isEmpty) {
+      return 'Can\'t be empty';
+    } else {
+      return null;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Colors.grey.shade900,
+      body: GestureDetector(
+        onTap: () {
+          FocusScope.of(context).requestFocus(FocusNode());
+        },
+        child: Stack(
+          children: [
+            SizedBox(
+              width: MediaQuery.of(context).size.width,
+              height: MediaQuery.of(context).size.height,
```

Comment:
O `MediaQuery` pode ser instanciado apenas uma vez, declarando uma váriavel `final size = MediaQuery.of(context).size` e usando o width e height da variável.

Reply:
Variáveis sizeWidth e sizeHeight criadas e usadas nas dimensões do SizedBox.